sled-agent/src/services.rs

```diff
@@ -4536,88 +4632,6 @@ mod test {
         logctx.cleanup_successful();
     }
 
-    #[tokio::test]
-    async fn test_old_ledger_migration_continue() {
```

Comment:
I think this test passed because of somewhat erroneous implementation details (see: my comments on the "three way merge of configs and zones" being way too complicated).

The set of zones in `tests/old-service-ledgers/rack2-sled10.json` contains many zones -- like CockroachDB, Crucible, etc -- that require explicit datasets to be present. We do nothing in the test to create those datasets. As a result, IMO, the call to `ensure_all_omicron_zones_persistent` below **should** be failing, because we cannot re-instantiate this configuration on a device missing those datasets.

That's what's happening now, but it undermines the value of this test, so I removed it.